src/Varnish.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Varnish\Varnish.
+ */
+
+namespace Netzstrategen\Varnish;
+
+/**
+ * Varnish client functionality.
+ */
+class Varnish {
+
+  /**
+   * URLs to purge in Varnish whose values are Booleans indicating regex.
+   *
+   * @var array
+   */
+  protected static $purgeUrls = [];
+
+  /**
+   * Purges a given URL from Varnish.
+   *
+   * @param string $url
+   *   The absolute URL to purge.
+   * @param bool $is_regex
+   *   Whether $url is a regular expression.
+   */
+  public static function purgeUrl($url, $is_regex = FALSE) {
+    $headers = array();
+
+    if ($is_regex) {
+      $headers['X-Purge-Method'] = 'regex';
+    }
+
+    $url_parts = parse_url($url);
+    $path = $url_parts['path'] ?? '';
+
+    $schema = defined('VARNISH_SCHEMA') ? VARNISH_SCHEMA : 'http://';
+
+    if (defined('VARNISH_IP')) {
+      $varnish_purge_url = $schema . VARNISH_IP . $path;
```

Comment:
A typical definition might include a port, which is optional, but possible, and not uncommon (whenever Varnish is hosted on the same machine as the webserver):
```php
const VARNISH_IP = '127.0.0.1:8080';
```
Not particularly happy with introducing this as an "IP" instead of e.g. "HOST".  Opinions?

Suggested change:
```php
const VARNISH_HOST = '127.0.0.1:8080';
```
and because the port is tightly coupled to the protocol:
```php
const VARNISH_HOST = 'http://127.0.0.1:8080';
```

_Background: Maxcluster supports to run Varnish on port 8000 (SSL) or 8080 (non-SSL), but even if it runs with SSL, the internal port for PURGE requests is 8080 (and thus also HTTP)._

This would additionally get rid of the separate `VARNISH_SCHEMA` constant which is optionally used just a few lines above.

Reply:
IMO using the `VARNISH_HOST` constant is a better approach as we can get rid of  another constant. As the port is coupled to the protocol anyway there should not be any downside, correct?